src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -215,6 +218,23 @@ def prepare_or_run_pipeline(
             )
             logger.info("Pipeline completed successfully.")
 
+    def _get_region_name(self) -> str:
+        """Returns the AWS region name.
+
+        Returns:
+            The region name.
+
+        Raises:
+            RuntimeError: If the region name cannot be retrieved.
+        """
+        try:
+            return cast(str, sagemaker.Session().boto_region_name)
+        except Exception as e:
+            raise RuntimeError(
```

Comment:
it should be a `logger.error(...)` now right? We dont want it to have an exception just because we couldnt get the region

Reply:
I'm catching the exception where it's called.

Reply:
You are catching it and then raising another exception so rather catch it and do a logger.error so that the program doesnt fail

Reply:
Hadn't pushed my code 🙈. In any case, I think we're agreed / aligned on what needs to happen here.